components/PortalStatistics.lua

```diff
@@ -0,0 +1,1407 @@
+
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{ConditionNode(ColumnName('birthdate'), Comparator.neq, '')}
+		:add{ConditionNode(ColumnName('birthdate'), Comparator.neq, EPOCH_DATE)}
+		:add{ConditionNode(ColumnName('deathdate'), Comparator.eq, EPOCH_DATE)}
+		:add{ConditionNode(ColumnName('earnings'), Comparator.gt, args.earnings)}
+
+	if Logic.readBool(args.isActive) then
+		conditions:add{ConditionNode(ColumnName('status'), Comparator.eq, 'Active')}
+	end
+
+	if Logic.readBool(args.playersOnly) then
+		local typeConditions = ConditionTree(BooleanOperator.any)
+		typeConditions:add{
+			ConditionNode(ColumnName('type'), Comparator.eq, 'player'),
+			ConditionNode(ColumnName('type'), Comparator.eq, 'Player'),
+		}
+		conditions:add{typeConditions}
+	end
+
+	local playerData = StatisticsPortal._getPlayers(args.limit, conditions:toString(), args.order)
+
+	local tbl = mw.html.create('table')
+		:addClass('wikitable wikitable-striped sortable')
+		:css('margin-left', '0px')
+		:css('margin-right', 'auto')
+
+	tbl:tag('tr')
+		:tag('th'):wikitext('ID'):addClass('unsortable'):done()
+		:tag('th'):wikitext('Age')
+
+	for _, player in ipairs(playerData) do
+		local birthdate =  DateExt.readTimestamp(player.birthdate) --[[@as integer]]
+		local age = os.date('*t', os.difftime(TIMESTAMP, birthdate))
+		local yearAge = age.year - 1970
+		local dayAge = age.yday - 1
+
+		tbl:tag('tr')
+			:tag('td')
+				:node(OpponentDisplay.BlockOpponent{
+					opponent = StatisticsPortal._toOpponent(player),
+					showPlayerTeam  = true,
+				}):done()
+			:tag('td')
+				:wikitext(yearAge .. ' years, ' .. dayAge .. ' days')
+	end
+
+	return mw.html.create('div'):addClass('table-responsive'):node(tbl)
+end
+
+
+--[[
+Section: Query Functions
+]]--
+
+
+---@param limit number?
+---@param addConditions string?
+---@param addOrder string?
+---@param addGroupBy string?
+---@return table
+function StatisticsPortal._getPlayers(limit, addConditions, addOrder, addGroupBy)
+	local data = mw.ext.LiquipediaDB.lpdb('player', {
+		query = 'pagename, id, nationality, earnings, extradata, birthdate, team',
+		conditions = addConditions or '',
+		order = addOrder,
+		groupby = addGroupBy,
+		limit = limit or MAX_QUERY_LIMIT,
+	})
+
+	return data
+end
+
+
+---@param limit number?
+---@param addConditions string?
+---@param addOrder string?
+---@param addGroupBy string?
+---@return table
+function StatisticsPortal._getTeams(limit, addConditions, addOrder, addGroupBy)
+	local data = mw.ext.LiquipediaDB.lpdb('team', {
+		query = 'pagename, name, template, earnings, extradata',
+		conditions = addConditions or '',
+		order = addOrder,
+		groupby = addGroupBy,
+		limit = limit or MAX_QUERY_LIMIT,
+	})
+
+	return data
+end
+
+
+---@param args table
+---@param groupBy string
+---@param defaultValue string
+---@param groupValues table
+---@return table
+function StatisticsPortal._getPieChartData(args, groupBy, defaultValue, groupValues)
+	args = args or {}
```

Comment:
```suggestion
```
as far as i can see the args table is always passed here after it is made sure it isn't nil